fvcore/common/checkpoint.py

```diff
@@ -296,22 +297,17 @@ def step(self, iteration: int, **kwargs: Any):
             self.checkpointer.save(
                 "model_{:07d}".format(iteration), **additional_state
             )
+            self.recent_checkpoints.append(
+                self.checkpointer.get_checkpoint_file()
+            )
 
-            if self.max_to_keep is not None:
-                all_checkpoint_files = (
-                    self.checkpointer.get_all_checkpoint_files()
-                )
-                all_checkpoint_files = [
-                    item
-                    for item in all_checkpoint_files
-                    if not item.endswith("model_final.pth")
-                ]
-                all_checkpoint_files.sort()
-                files_to_delete = all_checkpoint_files[: -self.max_to_keep]
-
-                for file in files_to_delete:
-                    if PathManager.exists(file):
-                        PathManager.rm(file)
+            if self.max_to_keep is not None and self.max_to_keep > 0:
```

Comment:
could you add `assert max_to_keep > 0` in `__init__`? It's a bit unclear whether `max_to_keep==0` means "do not keep anything" or "do not remove anything" so it's better to disable it.